src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,38 +1,65 @@
 package seedu.addressbook.data.person;
 
-import seedu.addressbook.data.exception.IllegalValueException;
+import java.util.Objects;
+
+import seedu.addressbook.data.person.address.Block;
+import seedu.addressbook.data.person.address.PostalCode;
+import seedu.addressbook.data.person.address.Street;
+import seedu.addressbook.data.person.address.Unit;
+import seedu.addressbook.data.tag.Tag;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address is entered in this format: a/BLOCK, STREET, UNIT, POSTAL_CODE";
 
-    public final String value;
+    private Block block;
+    private Street street;
+    private Unit unit;
+    private PostalCode postalCode;
     private boolean isPrivate;
-
+    public String value;
     /**
-     * Validates given address.
-     *
-     * @throws IllegalValueException if given address string is invalid.
+     * Assumption: Every field must be present and not null.
+     * Input entered in the order BLOCK, STREET, UNIT, POSTAL_CODE
      */
-    public Address(String address, boolean isPrivate) throws IllegalValueException {
+    public Address(Block block, Street street, Unit unit, PostalCode postalCode, boolean isPrivate) {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = address;
+        this.block = block;
+        this.street = street;
+        this.unit = unit;
+        this.postalCode = postalCode;
+        value = buildAddressString();
     }
-
-    /**
-     * Returns true if a given string is a valid person email.
-     */
-    public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    
+    public Address(String value, boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        this.value = value;
+        String[] Address = value.split(", ");
+        block = new Block(Address[0]);
+        street = new Street(Address[1]);
+        unit = new Unit(Address[2]);
+        postalCode = new PostalCode(Address[3]);
+    }
+    
+    public Block getBlock() {
```

Comment:
Because Block, Street are instance attributes in class Address, and they are private(encapsulated), so I wrote getter methods for them. Is this wrong?